packages/patterns/src/types.js

```diff
@@ -509,10 +509,10 @@ export {};
 
 /**
  * @template {Record<PropertyKey, MethodGuard>} [T=Record<PropertyKey, MethodGuard>]
- * @typedef {{ klass: 'Interface' } & InterfaceGuardPayload<T> } InterfaceGuard
- *
- * TODO https://github.com/endojs/endo/pull/1712 to make it into a genuine
- * guard that is distinct from a copyRecord
+ * @typedef {CopyTagged & {
+ *   [Symbol.toStringTag]: 'guard:interfaceGuard',
+ *   payload: InterfaceGuardPayload,
+ * }} InterfaceGuard
```

Comment:
I would somewhat prefer a definition of CopyTagged in `packages/pass-style/src/types.js` as a parameterized type:
```js
/**
 * @template {string} [Tag=string]
 * @template {Passable} [Payload=Passable]
 * @typedef {{
 *   [Symbol.toStringTag]: Tag,
 *   payload: Payload,
 *   [passStyle: symbol]: 'tagged' | Tag,
 * }} CopyTagged
 */
```

And then using here as:
```suggestion
 * @typedef {CopyTagged<'guard:InterfaceGuard', InterfaceGuardPayload>} InterfaceGuard
```

And ditto, for the other uses of CopyTagged.

Reply:
Trying now. But to clarify, changing the `[passStyle: symbol]` property to have a `| Tag` instead of a `| string` is wrong. Ideally it should have neither, since the only correct value is `'tagged'`. However, I cannot figure out how to do that without ts complaint. The `| string` seems to appease the checker while being least misleading of the options I tried. The comments says

```js
/**
 * ...
 * It must have a property with key equal to the `PASS_STYLE` export and value 'tagged'
 * and no other properties except `[Symbol.toStringTag]` and `payload`,
 * but TypeScript complains about a declaration like `[PASS_STYLE]: 'tagged'`
 * because importing packages do not know what `PASS_STYLE` is
 * so we appease it with a looser but less accurate definition
 * using symbol index properties.
 */
```

If you could help me figure out how to make this more accurate while still appeasing ts, that would be great, thanks.

Reply:
I like this *much* better, thanks!

Regarding the `| string`, while I would still like to clean this up, there's no reason to do so in this PR. Instead, I just clarified this comment and added a TODO.

Done